plotAfb.py

```diff
@@ -0,0 +1,287 @@
+							tempNom=tempHistPos.Clone()
+							tempDenom=tempHistPos.Clone()
+							tempNom.Add(tempHistNeg,-1)
+							tempDenom.Add(tempHistNeg)
+							tempDenom.Sumw2()
+							tempNom.Sumw2()
+							tempNom=tempNom.Rebin(len(bng) - 1, 'hist_' + uuid.uuid4().hex, array('d', bng))
+							tempDenom=tempDenom.Rebin(len(bng) - 1, 'hist_' + uuid.uuid4().hex, array('d', bng))
+							tempHistAfb=ROOT.TH1D("Afb"+flavor+year+category+dataType,"The distribution of Afb for "+year+" "+flavor+" result",len(binning(flavor))-1  ,array("f",binning(flavor)))
+							#print(tempNom.GetNbinsX())
+                                                        #print(tempDenom.GetNbinsX())
+							tempHistAfb.Divide(tempNom,tempDenom)
+							histTree[flavor][year][dataType][category]=tempHistAfb.Clone()			
+	for flavor in ["muon","electron"]:
+		for year in ["2016","2017","2018"]:
+			for category in ["BB","BE","All"]:
+				hCanvas = TCanvas("hCanvas", "Distribution", 500,500)
+				plotPad = ROOT.TPad("plotPad","plotPad",0,0,1,1)
+				setTDRStyle()	
+				plotPad.UseCurrentStyle()
+				plotPad.Draw()
+				plotPad.cd()
+				#plotPad.SetLogx()
+				xMax=histTree[flavor][year]["mc"][category].GetXaxis().GetXmax()
+				xMin=histTree[flavor][year]["mc"][category].GetXaxis().GetXmin()
+				yMax=histTree[flavor][year]["mc"][category].GetBinContent(histTree[flavor][year]["mc"][category].GetMaximumBin())
+				yMin=0.01
+				legend=ROOT.TLegend(0.1,0.7,0.4,0.98)
+				histTree[flavor][year]["data"][category].SetMarkerStyle(3)
+				histTree[flavor][year]["mc"][category].SetFillColorAlpha(2,0.5)
+				histTree[flavor][year]["mc"][category].SetTitle("Distribution of Afb for "+year+" "+flavor+" data")
+				histTree[flavor][year]["data"][category].GetXaxis().SetTitle("mll(GeV)")
+				histTree[flavor][year]["data"][category].GetYaxis().SetTitle("Afb")
+				histTree[flavor][year]["data"][category].GetYaxis().SetRangeUser(0.,1.)
+				legend.AddEntry(histTree[flavor][year]["data"][category],"Data","p")
+				legend.AddEntry(histTree[flavor][year]["mc"][category],"MC","f")
+				histTree[flavor][year]["data"][category].Draw()
+				histTree[flavor][year]["mc"][category].Draw("samehist")
+				legend.Draw()
+				#hCanvas.SetLogx()
+				hCanvas.Print("Distribution of Afb for "+year+" "+flavor+" data "+category+".pdf")
+	
+
+
+
+
+
+if __name__ == "__main__":
+	#parser.add_argument("-2016", "--2016", action="store_true", dest="use2016", default=False,
+						#help="use 2016 data and MC.")
+	#parser.add_argument("-2017", "--2017", action="store_true", dest="use2017", default=False,
+						#help="use 2017 data and MC.")
+	#parser.add_argument("-2018", "--2018", action="store_true", dest="use2018", default=False,
+                                                #help="use 2018 data and MC.")
+	os.chdir("/home/yang1452/CIAnalysis/plots")
+	backgrounds = ["Wjets","Other","DrellYan"]
```

Comment:
I think the WJets sample are a reason for the MC plots being to spikey, just remove it. We know that this background is flat in cos theta*